ntbs-service/DataMigration/ImportValidator.cs

```diff
@@ -89,48 +128,54 @@ public ImportValidator(IImportLogger logger, IReferenceDataRepository referenceD
+            var messages = new List<string>();
+            var lostData = new List<string>();
 
-            // Clean strings properties of this object's entity children
-            entity?
-                .GetType()
-                .GetProperties()
-                .Where(prop => Attribute.IsDefined(prop, typeof(ValidationChildAttribute)))
-                .ForEach(prop => TrimAndCleanStringProperties(prop.GetValue(entity)));
+            // The order we clean properties is important, because if we overwrite one value (eg. AdultsScreened) with
+            // null then we also need to overwrite the values that depend on it (eg. AdultsActiveTB)
+            var propertyNames = new List<string>
+            {
+                nameof(ContactTracing.AdultsScreened), nameof(ContactTracing.ChildrenScreened),
+                nameof(ContactTracing.AdultsActiveTB), nameof(ContactTracing.ChildrenActiveTB),
+                nameof(ContactTracing.AdultsLatentTB), nameof(ContactTracing.ChildrenLatentTB),
+                nameof(ContactTracing.AdultsStartedTreatment), nameof(ContactTracing.ChildrenStartedTreatment),
+                nameof(ContactTracing.AdultsFinishedTreatment), nameof(ContactTracing.ChildrenFinishedTreatment),
+            };
 
-            // Clean strings properties of this object's enumerable children
-            entity?
-                .GetType()
-                .GetProperties()
-                .Where(prop => Attribute.IsDefined(prop, typeof(ValidationChildEnumerableAttribute)))
-                .Select(prop => prop.GetValue(entity))
-                .Where(enumerable => enumerable != null)
-                .OfType<IEnumerable<object>>()
-                // Flatten
-                .SelectMany(element => element)
-                .ForEach(TrimAndCleanStringProperties);
-        }
+            foreach (var propertyName in propertyNames)
+            {
+                var property = notification.ContactTracing
+                    .GetType()
+                    .GetProperty(propertyName);
+                var propertyValue = property?.GetValue(notification.ContactTracing);
 
-        private static string TrimAndCleanString(string s) => s?.Trim().Replace("\t", " ");
+                var validationResults =
+                    ValidateProperty(notification.ContactTracing, propertyValue, propertyName).ToList();
+                if (validationResults.Any())
+                {
+                    property?.SetValue(notification.ContactTracing, null);
+                    messages.AddRange(validationResults.Select(result => result.ErrorMessage));
+                    lostData.Add($"{propertyName}: {propertyValue}");
+                }
+            }
+
+            foreach (var message in messages)
+            {
+                await _logger.LogNotificationWarning(context, runId, notification.LegacyId, message);
+            }
+
+            notification.ClinicalDetails.Notes += "\n" + string.Join(", ", lostData);
```

Comment:
I think the note that we're adding needs some context... e.g. "The following values were invalid and were removed from the notification"

Reply:
Yeah, definitely! Added